packages/web-components/@sfx/search-box/test/unit/search-box.test.ts

```diff
@@ -0,0 +1,239 @@
+import { SEARCHBOX_EVENT } from '../../src/utils';
+import { expect, spy, stub } from '../utils';
+import SearchBox from '../../src/search-box';
+import { TemplateResult, html, LitElement } from 'lit-element';
+
+describe('SearchBox Component', () => {
+  let searchbox;
+  let windowDispatchEvent;
+
+  beforeEach(() => {
+    searchbox = new SearchBox();
+    windowDispatchEvent = spy(window, 'dispatchEvent');
+  });
+
+  afterEach(() => {
+    windowDispatchEvent.restore();
+  });
+
+  /*
+   * To be replaced with Base
+   */
+  describe('Constructor', () => {
+    it('should extend LitElement', () => {
+      expect(searchbox).to.be.an.instanceof(LitElement);
+    });
+  });
+
+  describe('connectCallback', () => {
+    it('should add an autocomplete_hover eventListener to the window', () => {
+      const windowAddEventListener = spy(window, 'addEventListener');
+
+      searchbox.connectedCallback();
+
+      expect(windowAddEventListener).to.have.been.calledWith(
+        SEARCHBOX_EVENT.AUTOCOMPLETE_HOVER,
+        searchbox.updateText
+      );
+    });
+  });
+
+  describe('disconnectCallback', () => {
+    it('should remove the autocomplete_hover eventListener from the window', () => {
+      const windowRemoveEventListener = spy(window, 'removeEventListener');
+
+      searchbox.disconnectedCallback();
+
+      expect(windowRemoveEventListener).to.have.been.calledWith(
+        SEARCHBOX_EVENT.AUTOCOMPLETE_HOVER,
+        searchbox.updateText
+      );
+    });
+  });
+
+  describe('emitSearchEvent', () => {
+    it('should dispatch a search request event', () => {
+      searchbox.searchTerm = 'a';
+      const searchRequestEvent = new CustomEvent(
+        SEARCHBOX_EVENT.SEARCH_REQUEST,
+        {
+          detail: 'a',
+          bubbles: true
+        }
+      );
+
+      searchbox.emitSearchEvent();
+
+      expect(windowDispatchEvent).to.have.been.calledWith(searchRequestEvent);
+    });
+  });
+
+  describe('emitAutocompleteRequestEvent', () => {
+    it('should dispatch an autocomplete event', () => {
+      const autocompleteRequestEvent = new CustomEvent(
+        'sfx::autocomplete_request',
+        { detail: 'b', bubbles: true }
+      );
+
+      searchbox.emitAutocompleteRequestEvent();
+
+      expect(windowDispatchEvent).to.have.been.calledWith(
+        autocompleteRequestEvent
+      );
+    });
+  });
+
+  describe('emitSearchBoxClearedEvent', () => {
+    it('should dispatch an emitSearchBoxClearedEvent', () => {
+      const searchboxxClearedEvent = new CustomEvent('sfx::search_box_cleared');
+
+      searchbox.emitSearchBoxClearedEvent();
+
+      expect(windowDispatchEvent).to.have.been.calledWith(
+        searchboxxClearedEvent
+      );
+    });
+  });
+
+  describe('updateText', () => {
+    it('should update the searchTerm property in response to data received', () => {
+      const detail = 'inputText';
+      stub(searchbox, 'getInputElement').returns(
```

Comment:
I'm not sure this is what you'd want to do, but perhaps this test might go away with the other suggested changes. We can discuss this.

Reply:
Yes, this is definitely not what you want. `getInputElement` only needs to return an empty object for the test not to fail.

Reply:
Updated.